src/OceanSeaIceModels/InterfaceComputations/component_interfaces.jl

```diff
@@ -397,6 +397,9 @@ function ComponentInterfaces(atmosphere, ocean, sea_ice=nothing;
                                net_fluxes)
 end
 
+# To extend if we have a prognostic atmosphere
+state_exchanger(ocean, atmosphere) = StateExchanger(ocean, atmosphere)
```

Comment:
I had a different design --- ther eis an "`atmosphere_state_exchanger`" constructor. This is more composible, ie so you can mix and match different components. 

In other words we don't want to extend `StateExchanger`. We want to generalize `StateExchanger` to full generality, and then allow each component to interact with `StateExchanger` in their own way, with:

* `atmosphere_state_exchanger`
* `ocean_state_exchanger`
* `land_state_exchanger`
* `sea_ice_state_exchanger`

The `StateExchanger` is then the thing that is used to compute fluxes, ie on the exchange grid.